moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_objects.cpp

```diff
@@ -105,6 +105,70 @@ void MotionPlanningFrame::sceneScaleChanged(int value)
           s->scale((double)value / 100.0);
           ps->getWorldNonConst()->addToObject(scaled_object_->id_, shapes::ShapeConstPtr(s),
                                               scaled_object_->shape_poses_[i]);
+
+          // get shape bound size
+          double half_bound = -std::numeric_limits<double>::max();
+          switch(s->type){
+            case shapes::ShapeType::SPHERE:{
+              shapes::Sphere* sphere = static_cast<shapes::Sphere*>(s);
+              half_bound = sphere->radius;
+              break;
+            }
+            case shapes::ShapeType::CYLINDER:{
+              shapes::Cylinder* cylinder = static_cast<shapes::Cylinder*>(s);
+              half_bound = (cylinder->radius > cylinder->length/2.0) ? cylinder->radius : cylinder->length/2.0;
+              break;
+            }
+            case shapes::ShapeType::CONE:{
+              shapes::Cone* cone = static_cast<shapes::Cone*>(s);
+              half_bound = (cone->radius > cone->length/2.0) ? cone->radius : cone->length/2.0;
+              break;
+            }
+            case shapes::ShapeType::BOX:{
+              shapes::Box* box = static_cast<shapes::Box*>(s);
+              for(std::size_t i = 0; i < 3; i++)
+                half_bound = (half_bound < box->size[i]/2.0) ? box->size[i]/2.0 : half_bound;
+              break;
+            }
+            case shapes::ShapeType::MESH:{
+              shapes::Mesh* mesh = static_cast<shapes::Mesh*>(s);
+              for(std::size_t i = 0; i < mesh->vertex_count*3+3; i++)
+                half_bound = (half_bound < abs(mesh->vertices[i])) ? abs(mesh->vertices[i]) : half_bound;
+              break;
+            }
+            default:
+              break;
+          }
+          // Change marker size
+          for(std::size_t i = 0; i < viz_scene_marker_->controls.size(); i++)
+          {
+            for(std::size_t j = 0; j < viz_scene_marker_->controls[i].markers.size(); j++)
+            {
+              switch(viz_scene_marker_->controls[i].markers[j].type){
+                case visualization_msgs::Marker::ARROW:{
+                  // // bound + padding (40%) size
+                  double bound_size = half_bound*2.0*1.4;
+                  viz_scene_marker_->controls[i].markers[j].points[0].x = pow(-1.0,j%2)*(bound_size*0.5);
+                  viz_scene_marker_->controls[i].markers[j].points[1].x = pow(-1.0,j%2)*(bound_size*0.9);
+                  viz_scene_marker_->controls[i].markers[j].scale.x = bound_size*0.15;
+                  viz_scene_marker_->controls[i].markers[j].scale.y = bound_size*0.25;
+                  viz_scene_marker_->controls[i].markers[j].scale.z = bound_size*0.2;                  
```

Comment:
why different scaling factors for x, y, and z?

Reply:
For the `visualization_msgs/Marker` message, the `scale` is used different way for each marker types.
For this part, I am using `Arrow` marker. And on this type, 
- scale.x is the shaft diameter.
- scale.y is the head diameter
- scale.z the head length.
Thus, to keep same size ratio, I need to set different numbers for them.